src/vsr/replica.zig

```diff
@@ -8192,16 +8192,18 @@ pub fn ReplicaType(
             const reply = self.message_bus.get_message();
             defer self.message_bus.unref(reply);
 
-            const trailer_data = self.superblock.trailer_data(switch (command) {
-                .request_sync_manifest => .manifest,
-                .request_sync_free_set => .free_set,
-                .request_sync_client_sessions => .client_sessions,
-                else => unreachable,
-            });
-            assert(trailer_data.size >= parameters.offset);
+            const trailer = for (comptime std.enums.values(vsr.SuperBlockTrailer)) |trailer| {
+                if (command == SyncTrailer.requests.get(trailer)) break trailer;
+            } else unreachable;
+
+            const trailer_buffer_all = self.superblock.trailer_buffer(trailer);
+            const trailer_checksum = self.superblock.staging.trailer_checksum(trailer);
+            const trailer_size = self.superblock.staging.trailer_size(trailer);
+            assert(trailer_size >= parameters.offset);
```

Comment:
We repeat similarly-shaped bit of code in superblock. Maybe we need a pair of helpers? 

```
self.superblock.trailer_data(trailer) -> []u8
self.superblock.trailer_data_sector_aligned(trailer) -> [][constants.sector_size]u8
```

(unrelated, but, yeah, I wonder if it makes sense to enforce, at compile time, that write_sectors accepts some number of sectrors, and not just []u8

Reply:
Hmm it seems like all of the uses are just slightly different :thinking: 
1. replica here just wants the bytes within the data; doesn't care about alignment
2. `superblock.write_trailer` wants to round up to the nearest sector -- but I think handling that explicitly there is a good thing, since it needs to be careful to zero out the sector padding
3. `superblock.read_trailer` wants to round up to the nearest sector

I think I'll leave it as-is.